modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscription.java

```diff
@@ -312,8 +312,6 @@ public void closeConnection(UUID channelID, String nodeID) throws AndesException
                         + " on non durable subscription close");
             }
         }
-
-        this.subscriberConnection = null;
```

Comment:
CQEngine indexes subscriptions by connected node etc. It needs this to be not null at any time we query. 

when closing subscriptions

1. close connection -- > here is where above code made it null
2. remove from subscription registry (CQEngine access)
 
At step 2 it looks for the values. Thus keeping it until removal is not a problem as we remove the subscription everywhere  we close it.